### ISSUE TITLE
draw curve in analyze_logs.py will generate extra lines

### ISSUE THREAD

**Issue by foreverbest** [open-mmlab/mmsegmentation#715]
Lots of extra lines were drawn when utilizing analyze_logs.py to generate loss curve. 

- For example:

**command:**
python tools/analyze_logs.py work_dirs/101/deeplabv3plus/20210713_161930.log.json --keys loss --legend loss --out work_dirs/101/deeplabv3plus/loss
**output:**
plot curve of work_dirs/101/deeplabv3plus/20210713_161930.log.json, metric is loss
save curve to: work_dirs/101/deeplabv3plus/loss
![loss](https://user-images.githubusercontent.com/29807983/126582664-15ad57f2-6206-4626-aa8d-2873e2b1811c.png)

**Comment by xiexinch** [comment]
Hi @foreverbest 
Thanks for your feedback, we will fix log error soon. #694

**Comment by Junjun2016** [comment]
Hi @foreverbest 
Does it work for you?

**Comment by foreverbest** [comment]
Thanks for your great job, it works perfectly!

**Comment by hanzhy-code** [comment]
@xiexinch There is a question that whether the data is the loss of the validation set. They are not false data. This is my record:
...
{"mode": "train", "epoch": 5, "iter": 3950, "lr": 0.00912, "memory": 10356, "data_time": 0.00958, "decode.loss_seg": 0.05875, "decode.acc_seg": 59.21286, "aux.loss_seg": 0.02683, "aux.acc_seg": 59.25421, "loss": 0.08558, "time": 0.56193}

{"mode": "train", "epoch": 5, "iter":###  **640,** "lr": 0.0091, "memory": 10356, "aAcc": 0.9461, "mIoU": 0.4404, "mAcc": 0.4719, "IoU.\u80cc\u666f": 0.8956, "IoU.\u53ef\u884c\u9a76\u533a\u57df": 0.9124, "IoU.\u6c34\u5751": 0.0, "IoU.\u626c\u7070": 0.3942, "IoU.\u969c\u788d\u7269": 0.0, "Acc.\u80cc\u666f": 0.9285, "Acc.\u53ef\u884c\u9a76\u533a\u57df": 0.9746, "Acc.\u6c34\u5751": 0.0, "Acc.\u626c\u7070": 0.4565, "Acc.\u969c\u788d\u7269": 0.0, "data_time": 0.00951, "decode.loss_seg": 0.05932, "decode.acc_seg": 59.12221, "aux.loss_seg": 0.02716, "aux.acc_seg": 59.13662, "loss": 0.08648, "time": 0.55469}

{"mode": "train", "epoch": 5, "iter": 4050, "lr": 0.00909, "memory": 10356, "data_time": 0.00949, "decode.loss_seg": 0.06347, "decode.acc_seg": 58.92023, "aux.loss_seg": 0.02894, "aux.acc_seg": 58.96547, "loss": 0.09241, "time": 0.52901}
...

I set the evaluation interval as 4000,  and the size of my validation set is 2557, the batch size is 4, 2557/4=640. It caused the above error. So I want to know whether the data is the loss of the validation set.
